manifests/config.pp

```diff
@@ -21,37 +21,47 @@
     show_diff => false,
   }
 
-  $_add_user_keycloak_cmd = "${keycloak::install_base}/bin/add-user-keycloak.sh"
-  $_add_user_keycloak_args = "--user ${keycloak::admin_user} --password ${keycloak::admin_user_password} --realm master"
-  $_add_user_keycloak_state = "${keycloak::install_base}/.create-keycloak-admin-${keycloak::datasource_driver}"
-  exec { 'create-keycloak-admin':
-    command => "${_add_user_keycloak_cmd} ${_add_user_keycloak_args} && touch ${_add_user_keycloak_state}",
-    creates => $_add_user_keycloak_state,
-    notify  => Class['keycloak::service'],
-    user    => $keycloak::user,
-  }
-
   file { "${keycloak::install_base}/tmp":
     ensure => 'directory',
     owner  => $keycloak::user,
     group  => $keycloak::group,
     mode   => '0755',
   }
 
-  file { "${keycloak::install_base}/standalone/configuration":
-    ensure => 'directory',
-    owner  => $keycloak::user,
-    group  => $keycloak::group,
-    mode   => '0750',
+  $_add_user_keycloak_cmd = "${keycloak::install_base}/bin/add-user-keycloak.sh"
+  $_add_user_keycloak_state = "${keycloak::install_base}/.create-keycloak-admin-${keycloak::datasource_driver}"
+
+  if $::keycloak::operating_mode != 'domain' {
+    $_add_user_keycloak_args = "--user ${keycloak::admin_user} --password ${keycloak::admin_user_password} --realm master"
+    $_subdir = 'standalone'
+    $_config_cli_content = template('keycloak/config.cli.erb')
+    $_java_opts_path = "${keycloak::install_base}/bin/standalone.conf"
+  } else {
+    $_server_conf_dir = "${keycloak::install_base}/domain/servers/${keycloak::server_name}/configuration"
+    $_add_user_keycloak_args = "--user ${keycloak::admin_user} --password ${keycloak::admin_user_password} --realm master --sc ${_server_conf_dir}/" # lint:ignore:140chars
+    $_subdir = 'domain'
+    $_config_cli_content = template('keycloak/config-domain.cli.erb')
```

Comment:
Because this is an ERB template, it might make sense to stick with a single template for all modes and then just have something like this for the places that are specific to something like domain mode:

```
<%- if scope['keycloak::operating_mode'] == 'domain' -%>
// domain specific changes here
<%- end -%>
```

This will avoid a lot of duplication since the two templates do have a lot of overlap.

Reply:
:+1: